client/allocrunner/consul_hook_ce.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !ent
+// +build !ent
+
+package allocrunner
+
+import (
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/consul"
+	structsc "github.com/hashicorp/nomad/nomad/structs/config"
+)
+
+func getConsulClients(configs map[string]*structsc.ConsulConfig, logger hclog.Logger) (map[string]consul.Client, error) {
+	conf := configs["default"] // Nomad CE only supports a single Consul client
+
+	client, err := consul.NewConsulClient(conf, logger)
+
+	return map[string]consul.Client{"default": client}, err
+}
```

Comment:
It occurs to me that because we're returning a map of clients anyways, and the configs we get use CE/ENT-specific code to get the list of configs, you could just iterate over that map of configs to get the clients and safely end up with the right default for CE. And that would let you avoid having to have a separate `consul_hook_ce.go` and `consul_hook_ent.go`.

Reply:
addressed in c5ed8c9175e52566c9f35519df77e0516a808eb6